src/write_dump.cpp

```diff
@@ -50,15 +50,21 @@ void WriteDump::command(int narg, char **arg)
   int dumpfreq = MAX(1, update->nsteps);
   dumpfreq += update->ntimestep % dumpfreq;
 
+  std::string dump_id = "WRITE_DUMP";
   auto dumpargs = new char *[modindex + 2];
-  dumpargs[0] = (char *) "WRITE_DUMP";                      // dump id
+  dumpargs[0] = (char *) dump_id.c_str();                   // dump id
   dumpargs[1] = arg[0];                                     // group
   dumpargs[2] = arg[1];                                     // dump style
   dumpargs[3] = utils::strdup(std::to_string(dumpfreq));    // dump frequency
 
+  // delete existing dump with same ID to avoid re-use of dump-ID error
```

Comment:
I don't understand this comment, or possibly the bullet point in the PR summary about this change.  What is a scenario when a dump with the same ID already exists?  Possibly just a library interface issue?  What happens if the user script previoulsy defined a dump with the same ID - we wouldn't want to delete it, correct?

Reply:
The scenario can happen when you issue a write_dump command twice, but there is an error in the dump_modify part. You have to use the library interface to come across it (it happened to me when testing some feature in LAMMPS-GUI) so that exceptions are caught and thus the application doesn't crash instantly. The annoying part is that you cannot easily resolve this without explicitly using undump.

> What happens if the user script previoulsy defined a dump with the same ID - we wouldn't want to delete it, correct?

That is a tricky issue that applies to a lot of "internal" commands. Usually we derive the internal ID from the parent command/style. This doesn't exist in this special case.

The alternative would be to put a try/catch around the call to dump_modify and delete the dump if the dump_modify fails and then rethrow the error. That makes for rather messy output though.

Reply:
Here is a small reproducer script:
```python
#!/usr/bin/env python3

from lammps import lammps

lmp = lammps()
lmp.file('in.melt')
try:
  lmp.command('write_dump all image image.jpg type type modify adiam 1 0')
except:
  print("write_dump should fail so this is OK")
try:
  lmp.command('write_dump all image image.jpg type type modify adiam 1 1')
except:
  print("write_dump should not fail so this is not OK")
```
Without the change you get:
```
Dangerous builds not checked
ERROR: Illegal dump_modify command (src/dump_image.cpp:1562)
Last command: write_dump all image image.jpg type type modify adiam 1 0
write_dump should fail so this is OK
ERROR: Reuse of dump ID: WRITE_DUMP (src/output.cpp:748)
Last command: write_dump all image image.jpg type type modify adiam 1 1
write_dump should not fail so this is not OK
Total wall time: 0:00:00
```
But you *should* get:
```
Dangerous builds not checked
ERROR: Illegal dump_modify command (src/dump_image.cpp:1562)
Last command: write_dump all image image.jpg type type modify adiam 1 0
write_dump should fail so this is OK
Total wall time: 0:00:00
```

I'll try again with the try/catch. I had tried this before, but I see now a cleaner alternative.